language/move-binary-format/src/file_format.rs

```diff
@@ -251,6 +251,28 @@ pub struct StructHandle {
     pub abilities: AbilitySet,
     /// The type formals (identified by their index into the vec) and their constraints
     pub type_parameters: Vec<AbilitySet>,
+    /// The kind of each formal (either phantom or non-phantom)
+    pub type_param_kinds: Vec<TypeParamKind>,
+}
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+pub enum TypeParamKind {
+    NonPhantom,
+    Phantom,
+}
+
+impl TypeParamKind {
+    /// Returns `true` if the type_param_kind is [`Phantom`].
+    pub fn is_phantom(&self) -> bool {
+        matches!(self, Self::Phantom)
+    }
+
+    /// Returns `true` if the type_param_kind is [`NonPhantom`].
+    pub fn is_non_phantom(&self) -> bool {
+        matches!(self, Self::NonPhantom)
+    }
```

Comment:
I'd probably remove this function, it feels a bit confusing to me (and probably unused?)